charts/metering-ansible-operator/templates/crds/reportquery.crd.yaml

```diff
@@ -23,3 +23,103 @@ spec:
   - name: Age
     type: date
     JSONPath: .metadata.creationTimestamp
+
+  validation:
+  openAPIV3Schema:
+    required: ["spec"]
+    type: object
+    properties:
+      spec:
+        type: object
+        required: ["columns", "query"]
+        properties:
+          columns:
+            type: array
+            required: ["names"]
+            items:
+              type: object
+              required: ["type"]
+              properties:
+                name:
+                  type: string
+                  minLength: 1
+                type:
+                  type: string
+                  enum:
+                    - BOOLEAN
+                    - TINYINT
+                    - SMALLINT
+                    - INTEGER
+                    - BIGINT
+                    - REAL
+                    - DOUBLE
+                    - DECIMAL
+                    - VARCHAR
+                    - CHAR
+                    - VARBINARY
+                    - JSON
+                    - DATE
+                    - TIME
+                    - TIMESTAMP
+                    - ARRAY
+                    - MAP
+                    - MAP<VARCHAR, VARCHAR>
+                    - MAP<VARCHAR, INT>
+                    - MAP<INT, INT>
+                    - MAP<INT, VARCHAR>
+                    - ROW
+                    - IPADDRESS
+                    - UUID
+                    - HYPERLOGLOG
+                    - P4HYPERLOGLOG
+                    - QDIGEST
+                unit:
+                  type: string
+                  enum:
+                    - DATE
+                    - KUBERNETES_POD
+                    - KUBERNETES_PERSISTENTVOLUMECLAIM
+                    - KUBERNETES_PERSISTENTVOLUME
+                    - KUBERNETES_STORAGECLASS
+                    - KUBERNETES_NAMESPACE
+                    - BYTES
+                    - BYTE_SECONDS
+                    - TIME
+                    - CPU_CORE_SECONDS
+                    - CPU_CORES
+                    - MEMORY_BYTES
+                    - MEMORY_BYTES_SECONDS
+                    - SECONDS
+                tableHidden:
+                  type: boolean
+          inputs:
+            type: array
+            required: ["name"]
+            items:
+              type: object
+            properties:
+              name:
+                type: string
+                minLength: 1
+                type: boolean
+              type:
+                type: string
+                enum:
+                  - STRING
+                  - INTEGER
+                  - TIME
+                  - REPORTDATASOURCE
+                  - REPORTQUERY
+                  - REPORT
+              default:
+                type: string
+                oneOf:
```

Comment:
you shouldn't need to define `type: string` (line 115), then specify a list of compatible types (which also includes `type: string`)

Reply:
Apparently, we can't do `oneOf` and include `type: ...`, see: https://kubernetes.io/blog/2019/06/20/crd-structural-schema/
```
(i) inside of value validations no additionalProperties, type, nullable, title, description
...
inside of oneOf it is not allowed to use type (rule 3-i).
```

Reply:
We would just be able to do something like this: 
```
default: {}
```
This would mean that `default` can take any value/type as the keyword `AnyValue` is implicit.